src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -325,9 +341,212 @@ warning("PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrix
+
+            fin.close();
+        }
+        else
+        {
+            error("Cannot find Listmode cache on disk. Please recompute or comment out the  max cache size. Abort.");
+            return true;
+        }
+
+        info( boost::format("Cached Events: %1% ") % record_cache.size());
+        return false; // Stop here!!!
+    }
+
+    if(cache_lm_file)
+    {
+        info( boost::format("Listmode reconstruction: Serializing inputs ..."));
+        info( boost::format("Listmode reconstruction: Creating cache..."));
+
+        record_cache.reserve(cache_size);
+        additive_cache.reserve(cache_size);
+
+        this->list_mode_data_sptr->reset();
+        const shared_ptr<ListRecord> & record_sptr = this->list_mode_data_sptr->get_empty_record_sptr();
+        info(boost::format("Caching... "));
+
+        if(is_null_ptr(additive_proj_data_sptr))
+        {
+            while (true)
+            {
+                if(this->list_mode_data_sptr->get_next_record(*record_sptr) == Succeeded::no)
+                {
+                    break;
+                }
+
+                if (record_sptr->is_event() && record_sptr->event().is_prompt())
+                {
+                    Bin tmp;
+                    tmp.set_bin_value(1.0);
+                    record_sptr->event().get_bin(tmp, *proj_data_info_sptr);
+
+                    if (tmp.get_bin_value() != 1.0f
+                            ||  tmp.segment_num() < proj_data_info_sptr->get_min_segment_num()
+                            ||  tmp.segment_num()  > proj_data_info_sptr->get_max_segment_num()
+                            ||  tmp.tangential_pos_num() < proj_data_info_sptr->get_min_tangential_pos_num()
+                            ||  tmp.tangential_pos_num() > proj_data_info_sptr->get_max_tangential_pos_num()
+                            ||  tmp.axial_pos_num() < proj_data_info_sptr->get_min_axial_pos_num(tmp.segment_num())
+                            ||  tmp.axial_pos_num() > proj_data_info_sptr->get_max_axial_pos_num(tmp.segment_num())
+          #ifdef STIR_TOF
+                            ||  tmp.timing_pos_num() < proj_data_info_sptr->get_min_tof_pos_num()
+                            ||  tmp.timing_pos_num() > proj_data_info_sptr->get_max_tof_pos_num()
+          #endif
+                            )
+                    {
+                        continue;
+                    }
+                    record_cache.push_back(tmp);
+
+                    if (record_cache.size() > 1 && record_cache.size()%500000L==0)
+                        info( boost::format("Cached Prompt Events: %1% ") % record_cache.size());
+
+                    if(this->num_events_to_use > 0)
+                        if (record_cache.size() >= this->num_events_to_use)
+                            break;
+                }
+
+            }
+        }
+        else // With additive correction we should be interpolating the TOF positions.
+        {
+  #ifdef STIR_TOF
+
```

Comment:
missing code I guess, so something like
```suggestion
         // TODO
         error("listmode processing with caching is not yet supported for TOF");
```

Reply:
I have the code in my private repo. But I will commit it later.

Reply:
sure. I wanted to get this merged ASAP, so this quick fix will allow us to do that.